contracts/PrepaidCardMarketV2.sol

```diff
@@ -0,0 +1,375 @@
+pragma solidity ^0.8.9;
+pragma abicoder v1;
+
+import "./core/Ownable.sol";
+
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "./core/Versionable.sol";
+import "./IPrepaidCardMarket.sol";
+import "./PrepaidCardManager.sol";
+import "./ActionDispatcher.sol";
+import "./VersionManager.sol";
+import "./TokenManager.sol";
+import "./IPrepaidCardMarket.sol";
+import "./libraries/SafeERC677.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+contract PrepaidCardMarketV2 is
+  Ownable,
+  Versionable,
+  ReentrancyGuardUpgradeable,
+  IPrepaidCardMarket
+{
+  using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+  using SafeERC20Upgradeable for IERC677;
+  using SafeERC677 for IERC677;
+
+  struct SKU {
+    address issuerSafe;
+    address issuer;
+    address issuingToken;
+    uint256 faceValue;
+    string customizationDID;
+  }
+
+  address public prepaidCardManagerAddress;
+  address public tokenManager;
+  address public provisioner;
```

Comment:
why do we have both this and `trustedProvisioners`? aren't they both the same thing?

Reply:
(also i think a provisioner is assumed to be trusted--there is no such thing as an "untrusted provisioner", probably just calling this `provisioners` is fine).

Reply:
Actually I don't see a good reason to have both, unless we'll have many provisioners? If not we can remove the array and set the relayer as provisioner in tests

Reply:
Actually I don't see a good reason to have both, I think we can remove the `provisioner` and rename `trustedProvisioners` to `provisioners`

Reply:
Done!